File: jlpt/native_ios/Sources/Data/DTOs/StudyDTOs.swift
import Foundation

struct VocabularyStudyResponseDTO: Decodable {
    let item: VocabularyResponseDTO
    let status: StudyStatus
}

struct KanjiStudyResponseDTO: Decodable {
    let item: KanjiResponseDTO
    let status: StudyStatus
}

struct GrammarStudyResponseDTO: Decodable {
    let item: GrammarResponseDTO
    let status: StudyStatus
}

struct UpdateProgressRequestDTO: Encodable {
    let itemType: StudyItemType
    let itemId: Int64
    let status: StudyStatus
}

struct ReviewRequestDTO: Encodable {
    let itemType: StudyItemType
    let itemId: Int64
    let quality: Int
}

struct ReviewResultResponseDTO: Decodable {
    let status: StudyStatus
    let intervalDays: Int
    let nextReviewAt: String?
}

struct ProgressSummaryResponseDTO: Decodable {
    let countsByItemTypeAndStatus: [String: [String: Int64]]
}

File: jlpt/native_android/app/src/main/java/com/jlpt/android/data/remote/dto/StudyDtos.kt
package com.jlpt.android.data.remote.dto

import com.jlpt.android.domain.model.StudyItemType
import com.jlpt.android.domain.model.StudyStatus
import kotlinx.serialization.Serializable

@Serializable
data class VocabularyStudyResponseDto(
    val item: VocabularyResponseDto,
    val status: StudyStatus
)

@Serializable
data class KanjiStudyResponseDto(
    val item: KanjiResponseDto,
    val status: StudyStatus
)

@Serializable
data class GrammarStudyResponseDto(
    val item: GrammarResponseDto,
    val status: StudyStatus
)

@Serializable
data class UpdateProgressRequestDto(
    val itemType: StudyItemType,
    val itemId: Long,
    val status: StudyStatus
)

@Serializable
data class ReviewRequestDto(
    val itemType: StudyItemType,
    val itemId: Long,
    val quality: Int
)

@Serializable
data class ReviewResultResponseDto(
    val status: StudyStatus,
    val intervalDays: Int,
    val nextReviewAt: String? = null
)

@Serializable
data class ProgressSummaryResponseDto(
    val countsByItemTypeAndStatus: Map<StudyItemType, Map<StudyStatus, Long>> = emptyMap()
)
